common-rest/src/main/java/org/uniprot/api/common/repository/stream/rdf/RdfEntryCountProvider.java

```diff
@@ -0,0 +1,50 @@
+package org.uniprot.api.common.repository.stream.rdf;
+
+import java.util.Arrays;
+import java.util.Map;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class RdfEntryCountProvider {
+    private static final Map<String, Map<String, String>> regexMap =
+            Map.of(
+                    "uniprotkb",
+                            Map.of(
+                                    "rdf",
+                                            "<rdf:Description rdf:about=\"+([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])(-[0-9]+)?+\">",
+                                    "ttl",
+                                            "<([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])(-[0-9]+)?> rdf:type up:Protein ;",
+                                    "nt",
+                                            "<http://purl.uniprot.org/uniprot/([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])(-[0-9]+)?> <http://purl.uniprot.org/core/reviewed>"),
+                    "uniparc",
+                            Map.of(
+                                    "rdf", "<rdf:Description rdf:about=\"+UPI[\\w]{10}+\">",
+                                    "ttl", "<UPI[\\w]{10}> rdf:type up:Protein ;",
+                                    "nt",
+                                            "<http://purl.uniprot.org/uniprot/UPI[\\w]{10}> <http://purl.uniprot.org/core/reviewed>"),
+                    "uniref",
+                            Map.of(
+                                    "rdf",
+                                            "<rdf:Description rdf:about=\"+(UniRef100|UniRef90|UniRef50)_([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])(-[0-9]+)?+\">",
+                                    "ttl",
+                                            "<(UniRef100|UniRef90|UniRef50)_([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])(-[0-9]+)?> rdf:type up:Protein ;",
+                                    "nt",
+                                            "<http://purl.uniprot.org/uniprot/(UniRef100|UniRef90|UniRef50)_([OPQ][0-9][A-Z0-9]{3}[0-9]|[A-NR-Z]([0-9][A-Z][A-Z0-9]{2}){1,2}[0-9])(-[0-9]+)?> <http://purl.uniprot.org/core/reviewed>"));
+
```

Comment:
Wow. Map of map. 👍 But i see a problem with regex matching. When you call matches method in isAMatch, Java compiles the regex and which is kind of heavy operation. If you are calling the same regex multiple times and Java compiles each time, it is inefficient.
Change the value of Map to Pattern instead of String and call pattern.matcher().matches(str) instead of str.matches(regex). Search `Pattern.compile` in our codebase for example.

Reply:
Feel free to file a separate ticket to move all the regex at one common place. It would be easy maintain if we have one copy of regex for each data type.